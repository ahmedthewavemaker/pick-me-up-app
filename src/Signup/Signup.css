
.sign{
    display: flex;
    justify-content: center;
}

.forma{
    font-size: 25px;
}
.button{
    background-color: rebeccapurple;
    font-size: 20px;
}