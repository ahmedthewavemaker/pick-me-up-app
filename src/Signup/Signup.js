import React, {useState} from 'react'
import AuthApiService from '../services/auth-service'
import './Signup.css'




function Signup(props) {

    const [error, setError] = useState(null)

    const handleSubmit = ev => {
        ev.preventDefault()
        const {  user_name, password, full_name, location  } = ev.target
    
        setError(null)
        AuthApiService.postUser({
          user_name: user_name.value,
          password: password.value,
          full_name: full_name.value,
         location: location.value,
        })
          .then(user => {
            user_name.value = ''
            password.value = ''
            full_name.value = ''
            location.value = ''
            props.history.push('/games')
            
          })
          .catch(res => {
              console.log(res) 
           setError(res)
          })
      }

    return (
        <div className='sign'>
            <form onSubmit={handleSubmit} className='forma'>
                {error}
                <label htmlFor='user_name'> Username: </label>
                <input placeholder='email@site.com' type='text' name='user_name' id='user_name' />
                <br/>
                <label htmlFor='password'> Password: </label>
                <input placeholder='password' type='text' name='password' id='password' />
                <br/>
                <label htmlFor='full_name'> Nickame: </label>
                <input placeholder='name' type='text' name='full_name' id='full_name' />
                <br/>
                <label htmlFor='location'> Location: </label>
                <input placeholder='city, state' type='text' name='location' id='location' />
                <br/>
                <button className='button'>  Submit </button>
            </form>
        </div>


    )
}

export default Signup;