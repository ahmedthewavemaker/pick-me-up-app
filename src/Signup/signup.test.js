import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Signup from './Signup';
import { MemoryRouter } from 'react-router-dom';

describe('Signup component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <Signup />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <Signup  />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


});