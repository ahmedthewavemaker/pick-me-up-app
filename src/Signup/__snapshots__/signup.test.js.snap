// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="sign"
  >
    <form
      className="forma"
      onSubmit={[Function]}
    >
      <label
        htmlFor="user_name"
      >
         Username: 
      </label>
      <input
        id="user_name"
        name="user_name"
        placeholder="email@site.com"
        type="text"
      />
      <br />
      <label
        htmlFor="password"
      >
         Password: 
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        type="text"
      />
      <br />
      <label
        htmlFor="full_name"
      >
         Name: 
      </label>
      <input
        id="full_name"
        name="full_name"
        placeholder="name"
        type="text"
      />
      <br />
      <label
        htmlFor="location"
      >
         Location: 
      </label>
      <input
        id="location"
        name="location"
        placeholder="city, state"
        type="text"
      />
      <br />
      <button
        className="button"
      >
          Submit 
      </button>
    </form>
  </div>,
]
`;
